src/Pack.js

```diff
@@ -0,0 +1,194 @@
+/**
+    @external Viz
+    @see https://github.com/d3plus/d3plus-viz#Viz
+*/
+import {nest} from "d3-collection";
+import {hierarchy, pack} from "d3-hierarchy";
+
+import {accessor, assign, configPrep, constant, elem} from "d3plus-common";
+import * as shapes from "d3plus-shape";
+import {Viz} from "d3plus-viz";
+
+/**
+    @class Pack
+    @extends Viz
+    @desc Uses the [d3 pack layout](https://github.com/d3/d3-hierarchy#pack) to creates Circle Packing chart based on an array of data.
+*/
+export default class Pack extends Viz {
+
+  /**
+      @memberof Pack
+      @desc Invoked when creating a new class instance, and sets any default parameters.
+      @private
+  */
+  constructor() {
+
+    super();
+
+    this._layoutPadding = 1;
+    this._on.mouseenter = () => {};
+    this._on["mouseleave.shape"] = () => {
+      this._tooltip = false;
+      this.hover(false);
+    };
+
+    let hoverData = [];
+    const recursionCircles = d => {
+      if (d.values) {
+        d.values.forEach(h => {
+          hoverData.push(h);
+          recursionCircles(h);
+        });
+      }
+      else {
+        hoverData.push(d);
+      }
+    };
```

Comment:
1. move `recursionCircles` outside of the constructor and the Class
2. add a second argument to the function, which will act as hoverData, and have it default to empty (`(d, arr = [])`)
3. return that array as your recursively use the function (allowing you to remove the global `hoverData` variable entirely)